airgun/widgets.py

```diff
@@ -2090,3 +2091,42 @@ class Link(Text):
     def fill(self, value):
         if value:
             self.browser.click(self)
+
+
+class PopOverWidget(GenericLocatorWidget):
+    """Popover-content UI widget which contains header, drop_down or input_box and
+    submit button. This is associated within a table.
+
+    Example html representation::
+
+        <h3 class="popover-title">Authorize login delegation (Default: false)</h3>
+            <form class="form-inline editableform" style="">
+                <select class="form-control input-sm">
+                    <option value="0">No</option>
+                    <option value="1">Yes</option></select>
+                <div class="editable-buttons">
+                    <button type="submit" class="btn btn-primary btn-sm editable-submit">
+
+        <h3 class="popover-title">Authorize login delegation auth ...(Default: Not set)</h3>
+            <form class="form-inline editableform" style="">
+                <input type="text" ...>
+                <div class="editable-buttons">
+                    <button type="submit" class="btn btn-primary btn-sm editable-submit">
+
+    Locator example::
+
+        //div[contains(@class, 'editable-open')]
+    """
+
+    header = Text("//h3[@class='popover-title']")
+    input_box = TextInput(locator="//div[@class='editable-input']/input")
```

Comment:
I feel that `TextInput(locator="//div[@class='editable-input']/input")` will not match: `<div class="editable-input"><textarea class="form-control input-large" placeholder="fqdn,hostgroup,os,domain" rows="7"></textarea></div>`
We need to be able fill both inputs and textareas here

Reply:
nice catch, I  saw that but somehow missed it.